src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -177,1327 +205,1182 @@ static std::vector<rct_widget> _window_editor_object_selection_widgets = {
-                log_warning("Wrong sort type %d, leaving list as-is.", _listSortType);
+
+            case WIDX_ADVANCED:
+                list_information_type ^= 1;
+                Invalidate();
                 break;
-        }
-        if (sortFunc != nullptr)
-        {
-            std::sort(_listItems.begin(), _listItems.end(), sortFunc);
-            if (_listSortDescending)
+
+            case WIDX_INSTALL_TRACK:
             {
-                std::reverse(_listItems.begin(), _listItems.end());
+                if (selected_list_item != -1)
+                {
+                    selected_list_item = -1;
+                }
+                Invalidate();
+
+                auto intent = Intent(WC_LOADSAVE);
+                intent.putExtra(INTENT_EXTRA_LOADSAVE_TYPE, LOADSAVETYPE_LOAD | LOADSAVETYPE_TRACK);
+                context_open_intent(&intent);
+                break;
             }
+            case WIDX_FILTER_TEXT_BOX:
+                window_start_textbox(this, widgetIndex, STR_STRING, _filter_string, sizeof(_filter_string));
+                break;
+            case WIDX_FILTER_CLEAR_BUTTON:
+                std::fill_n(_filter_string, sizeof(_filter_string), 0x00);
+                FilterUpdateCounts();
+                scrolls->v_top = 0;
+                VisibleListRefresh();
+                Invalidate();
+                break;
+            case WIDX_LIST_SORT_TYPE:
+                if (_listSortType == RIDE_SORT_TYPE)
+                {
+                    _listSortDescending = !_listSortDescending;
+                }
+                else
+                {
+                    _listSortType = RIDE_SORT_TYPE;
+                    _listSortDescending = false;
+                }
+                VisibleListRefresh();
+                break;
+            case WIDX_LIST_SORT_RIDE:
+                if (_listSortType == RIDE_SORT_RIDE)
+                {
+                    _listSortDescending = !_listSortDescending;
+                }
+                else
+                {
+                    _listSortType = RIDE_SORT_RIDE;
+                    _listSortDescending = false;
+                }
+                VisibleListRefresh();
+                break;
+            default:
+                if (widgetIndex >= WIDX_TAB_1
+                    && static_cast<size_t>(widgetIndex) < WIDX_TAB_1 + std::size(ObjectSelectionPages))
+                {
+                    SetPage(widgetIndex - WIDX_TAB_1);
+                }
+                break;
         }
     }
-    w->Invalidate();
-}
 
-static void WindowEditorObjectSelectionInitWidgets()
-{
-    auto& widgets = _window_editor_object_selection_widgets;
-    if (!_window_editor_object_selection_widgets_initialised)
+    void OnResize() override
     {
-        _window_editor_object_selection_widgets_initialised = true;
-        auto tabWidget = widgets[widgets.size() - 2];
-        for (size_t i = 1; i < std::size(ObjectSelectionPages); i++)
+        min_width = 750;
```

Comment:
Revert this back to `window_set_size`

Reply:
Done https://github.com/OpenRCT2/OpenRCT2/pull/16027/commits/0460b8287005ce74f16eb1cfbf49a2c3ae80272c